core/client/ui/report.js

```diff
@@ -0,0 +1,115 @@
+  if (reported._id === Meteor.userId()) {
+    lp.notif.error('You can\'t report yourself');
+    return;
+  }
+  const isAdmin = getUser(Meteor.userId()).roles?.admin;
+
+  const text = document.querySelector(inputSelector).value;
+  const currentLevel = Meteor.user().profile.levelId;
+  if (!text) {
+    lp.notif.error('Please enter a message');
+    return;
+  }
+  if (text.length > 1024) {
+    lp.notif.error('Message is too long');
+    return;
+  }
+  let admins = getAllAdmins();
+  if (isAdmin) {
+    admins = admins.filter(admin => admin._id !== Meteor.userId());
+  }
+  if (!admins.length) {
+    // Quiting even if the user is an admin because we don't want to create a conversation with himself
+    lp.notif.error('No admins found');
+    return;
+  }
+  const adminsLevel = admins.filter(admin => admin.profile.levelId === currentLevel);
+  if (adminsLevel.length) {
+    admins = adminsLevel;
+  }
+  const level = Levels.findOne(currentLevel).name;
+  const message = buildMessage(text, level, reason, reported, messageId);
+  const channel = [...admins.map(admin => admin._id), Meteor.userId()].sort().join(';');
+
+  sendReport(channel, message);
+
+  closeConsole();
+};
+
+Template.report.onCreated(function () {
+  this.reason = new ReactiveVar('Non-specified');
+});
+
+Template.report.helpers({
+  title() {
+    const template = Template.instance();
+    return `Report user - ${getUser(template.data.userId)?.profile.name}`;
+  },
+  reportAllReasons() {
+    const reportAllReasons = ['Non-specified', 'Spam', 'Abusive chat', 'Cheating', 'Offensive name', 'Other'];
+    return reportAllReasons;
+  },
+  explanation() {
+    switch (Template.instance().reason.get()) {
+      case 'Non-specified':
+        return '';
+      case 'Spam':
+        return 'Spam is the use of the messages to send too many messages in a short period of time, to send the same message multiple times, or to repeatedly connect and disconnect with an user';
+      case 'Abusive chat':
+        return 'Abusive chat is the use of the messages to harass, insult, or threaten another user';
+      case 'Cheating':
+        return 'Cheating is the use of any third-party software or any other method to gain an unfair advantage in the game';
+      case 'Offensive name':
+        return 'Offensive name is the use of a name that is offensive, vulgar, or inappropriate';
+      case 'Other':
+        return 'Other behavior that is not listed above, please specify';
+      default:
+        return '';
+    }
+  },
+});
+
+Template.report.events({
+  'change .js-report-reason-select'(event) {
+    Template.instance().reason.set(event.target.value);
+  },
+  'click .js-button-submit'(event) {
+    event.preventDefault();
+    event.stopPropagation();
+    const template = Template.instance();
+    lp.notif.confirm('Report user', `Do you really want to report this message?`, () => onSubmit(template.reason.get(), getUser(template.data.userId), template.data.messageId));
```

Comment:
Does it say "this message" when we report a user through the radial menu directly?